### PR TITLE
Add test to exercise lookup corner cases

### PR DESCRIPTION
**Pull request by cce**
## Summary

This test attempts to cover the case when an accountUpdates.lookupX method can't find the requested address, falls through looking at deltas and the LRU accounts cache, then hits the database — only to discover that the round stored in the database (committed in `accountUpdates.commitRound`) is out of sync with `accountUpdates.cachedDBRound` (updated a little bit later in `accountUpdates.postCommit`).

In this case, the lookup method waits and tries again, iterating the `for { }` it is in. We did not have coverage for this code path before.

## Test Plan

Adds new test.

### CODE REVIEW

File: ledger/acctupdates_test.go [hunk 1]
```diff
@@ -1700,3 +1700,145 @@ func TestConsecutiveVersion(t *testing.T) {
 		protocol.ConsensusV21,
 	}
 }
+
+func TestAcctUpdatesLookupRetry(t *testing.T) {
```
Comment (algorandskiy) [inline ledger/acctupdates_test.go:1704]: Please add a comment explaining this non-trivial test

File: ledger/acctupdates_test.go [hunk 2]
```diff
@@ -1700,3 +1700,145 @@ func TestConsecutiveVersion(t *testing.T) {
...
+		checkAcctUpdates(t, au, 0, i, accts, rewardsLevels, proto)
+	}
+
+	flushRound := func(i basics.Round) {
+		// Clear the timer to ensure a flush
+		ml.trackers.lastFlushTime = time.Time{}
+
+		ml.trackers.committedUpTo(basics.Round(proto.MaxBalLookback) + i)
+		ml.trackers.waitAccountsWriting()
+		//checkAcctUpdates(t, au, i, basics.Round(proto.MaxBalLookback+14), accts, rewardsLevels, proto)
```
Comment (algorandskiy) [inline ledger/acctupdates_test.go:1788]: remove

File: ledger/acctupdates_test.go [hunk 3]
```diff
@@ -1700,3 +1700,145 @@ func TestConsecutiveVersion(t *testing.T) {
...
+		time.Sleep(200 * time.Millisecond)
+		stallingTracker.postCommitReleaseLock <- struct{}{}
+	}()
+
+	// issue a LookupWithoutRewards while persistedData.round != au.cachedDBRound
+	d, validThrough, err := au.LookupWithoutRewards(rnd, addr)
+	require.NoError(t, err)
+	require.Equal(t, d, data)
+	require.GreaterOrEqualf(t, uint64(validThrough), uint64(rnd), "validThrough: %v rnd :%v", validThrough, rnd)
+	t.Log("validThrough", validThrough)
```
Comment (algorandskiy) [inline ledger/acctupdates_test.go:1839]: remove

**Review by algorandskiy — COMMENTED** [review]
Looks good, couple minor notes

**Review by algorandskiy — APPROVED** [review]

**Review by tsachiherman — APPROVED** [review]
I'm a bit worried about random failures, since the Sleep(200ms) might create a non-consistent behavior on slow machines. But I think that this is a good starting point.

**Pull request merged**